core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -230,6 +228,61 @@ oppia.factory('TopicUpdateService', [
           // Get the current change list.
           var currentChangeList = UndoRedoService.getChangeList();
           var indicesToDelete = [];
+          var oldSubtopicIdSkillIdMap = {};
+          var index = null;
+          var skillIds = subtopic.getSkillIds();
+          for (var i = 0; i < currentChangeList.length; i++) {
+            var changeDict =
+              currentChangeList[i].getBackendChangeObject();
+            if (oldSubtopicIdSkillIdMap[changeDict.skill_id] === undefined) {
+              if (changeDict.new_subtopic_id === subtopicId) {
+                // Record the origin subtopic id for the skill.
+                oldSubtopicIdSkillIdMap[changeDict.skill_id] =
+                  changeDict.old_subtopic_id;
+                index = i;
+              }
+            } else {
+              if (changeDict.cmd === CMD_REMOVE_SKILL_ID_FROM_SUBTOPIC) {
+                // If the skill was removed from all subtopics at some point,
+                // modify that change dict, so that it was as if it was removed
+                // from its recorded origin.
+                changeDict.subtopic_id =
+                  oldSubtopicIdSkillIdMap[changeDict.skill_id];
```

Comment:
Should we also update the mapping, in case more stuff is done to that skill in the future?

Reply:
But, what I am trying to do is to only record the initial point of origin of a skill, and modify the last move related to that skill to have that point of origin, so I don't think oldSkillIdToSubtopicIdMapping need to modified, right? As all intermediate moves are ignored anyways (unless if it's to uncategorized, which is taken care of).

Reply:
Removed this.